src/Entity/Article.php

```diff
@@ -38,9 +38,18 @@ class Article
      */
     private $draft;
 
-    public function __construct()
+    private function __construct(string $title, string $content, string $slug)
     {
+        $this->content = $content;
         $this->dateCreation = new DateTimeImmutable();
+        $this->draft = true;
+        $this->slug = $slug;
+        $this->title = $title;
+    }
+
+    public static function write(string $title, string $content, string $slug): self
+    {
+        return new self($title, $content, $slug);
     }
```

Comment:
I read somewhere that it was more readable, in a business logic, to have `Article::write()` rather than 'new Article()' because "creating" and article has little meaning, its part of the abstraction glossary we created with the RAD paradigm.

Do you agree ?

Reply:
I totally agree, as long as it makes sense, semantically speaking. BTW, this is called a _named constructor_, and it’s an important part of DDD. I would not use this systematically though, but only when it’s appropriate, as it’s the case here.

It’s very true that every time you write some part of your domain model, you should think more about the semantics of your domain rather than the semantics of the programming language being used. Sticking to the semantics of your domain, in DDD, basically means: «How would the user describe that action, in their own words?», and indeed, in this context it makes more sense to «write an article» rather than «create a new article».